packages/contracts/contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol

```diff
@@ -971,75 +973,97 @@ contract OVM_CanonicalTransactionChain is iOVM_CanonicalTransactionChain, Lib_Ad
     }
 
     /**
-     * Checks that a given batch context has a time context which is below a given que element
-     * @param _context The batch context to validate has values lower.
-     * @param _queueIndex Index of the queue element we are validating came later than the context.
+     * For a fair of
```

Comment:
I'd suggest something like: `checks that, for a given pair of adjacent batch contexts: 1) the two contexts' relative blocknumber and timestamps are increasing, and 2) the relative blocknumber and timestamp of the earlier context's subsequent queue transactions fit between those values.`

Reply:
after reading the code below, there is also a third check about the force inclusion period which we probably want to reflect too.

Reply:
whoops, reading on again, the `1)` above is actually not handled here but in `_validateConsecutiveBatchContexts`.  I like the docstring there--monotonicity, nice and simple :)